pkg/jobrunaggregator/jobrunaggregatoranalyzer/pass_fail.go

```diff
@@ -203,9 +206,78 @@ func (a *weeklyAverageFromTenDays) getAggregatedTestRuns(ctx context.Context) (m
 	return a.aggregatedTestRunsByName, a.queryTestRunsErr
 }
 
+func getMajor(in string) int {
+	major, err := strconv.ParseInt(strings.Split(in, ".")[0], 10, 32)
+	if err != nil {
+		panic(err)
+	}
+	return int(major)
+}
+
+func getMinor(in string) int {
+	minor, err := strconv.ParseInt(strings.Split(in, ".")[1], 10, 32)
+	if err != nil {
+		panic(err)
+	}
+	return int(minor)
+}
+
+func (a *weeklyAverageFromTenDays) getFallBackJobName(ctx context.Context, jobName string) (string, error) {
+	allJobs, err := a.bigQueryClient.ListAllJobs(ctx)
+	if err != nil {
+		return jobName, err
+	}
+	var targetFromRelease, targetToRelease string
+	var job *jobrunaggregatorapi.JobRow
+	for _, j := range allJobs {
+		if j.JobName == jobName {
+			job = &j
+			break
+		}
+	}
+	if job != nil {
+		if len(job.FromRelease) > 0 {
+			fromReleaseMajor := getMajor(job.FromRelease)
+			fromReleaseMinor := getMinor(job.FromRelease)
+			targetFromRelease = fmt.Sprintf("%d.%d", fromReleaseMajor, fromReleaseMinor-1)
+		}
+		if len(job.Release) > 0 {
+			toReleaseMajor := getMajor(job.Release)
+			toReleaseMinor := getMinor(job.Release)
+			targetToRelease = fmt.Sprintf("%d.%d", toReleaseMajor, toReleaseMinor-1)
+		}
+
+		for _, j := range allJobs {
+			if j.Architecture == job.Architecture &&
+				j.Topology == job.Topology &&
+				j.Network == job.Network &&
+				j.Platform == job.Platform &&
+				j.FromRelease == targetFromRelease &&
+				j.Release == targetToRelease &&
+				j.IPMode == job.IPMode {
+				return j.JobName, nil
+			}
+		}
```

Comment:
Just curious if we have tested this / know that the match covers everything like  RT

periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-rt-upgrade

Any chance we could swap an RT job with 

periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-upgrade

Reply:
Good question. Job table does not have anything to differentiate RT vs non-RT jobs other than just the name. This mean, when we fall back, we might pick the other variant (RT fall back to non-RT or non-RT fall back to RT). Will that be a problem?

Reply:
Technically the values could be different.  I was wondering if we could scrub the original & target release versions from the job names and then do a compare, everything should match at that point.

Reply:
My only comment here is that is very specific to RT, don't know if that is the only case where a setting is captured in the job name but not the variants.  Just a thought.

Reply:
Long term plan is to have RT as a new variant. Name based matching is only an exception.

Reply:
Looks to be more than just RT based on 

select count (distinct(JobName)), Platform, Network, IPMode, Topology, Release, FromRelease, IsUpgrade FROM `openshift-ci-data-analysis.ci_data.UnifiedJobRuns`  where Release = '4.14' group by Cluster, Platform, Network, IPMode, Topology, Release, FromRelease, IsUpgrade

and

select distinct(JobName) FROM `openshift-ci-data-analysis.ci_data.UnifiedJobRuns` where Platform = 'aws' and Network = 'sdn' and IPMode = 'ipv4' and Topology = 'ha' and Release = '4.14' and FromRelease = '' and IsUpgrade = false

Will review further with Ken

Reply:
Patch to consider.

[ci-tools-fallback-normalize-jobname.patch.txt](https://github.com/openshift/ci-tools/files/11017851/ci-tools-fallback-normalize-jobname.patch.txt)

Reply:
Thanks! Took that.